frontends/verilog/verilog_parser.y

```diff
@@ -151,7 +151,9 @@ static void enterTypeScope()
 
 static void exitTypeScope()
 {
+	auto types = user_type_stack.back();
 	user_type_stack.pop_back();
+	delete types;
```

Comment:
There is a comment I recently added in `verilog_frontend.cc:add_package_types` discussing this leak that should probably be updated.

Reply:
Something I just thought of: should `design -reset[-vlog]` drop the type stack as well?

Reply:
Yes, I agree that it should.

Reply:
The whole user type stack handling actually needed a bit more work, I redid it in #2688